webidls/Browser.webidl

```diff
@@ -0,0 +1,115 @@
+/**
+  * Browser is the equivalent of a top level Constellation::Frame.
+  * Browser is a top level browsing context.
+  * ~= Tab, ~= Webview
+  */
+
+[Constructor(DOMString contextName, StorageSession storageSession)]
+interface Browser {
+  // FIXME: how is this attached to a viewport? For swapping, it's important to be able to attach and detach
+
+  Promise<void> setHandler(BrowserHandler handler);
+
+  // Used to expose special JS APIs.
+  // For example, can be used to expose the Browser API or Web Extensions APIs for content
+  // See: https://developer.chrome.com/extensions/content_scripts#execution-environment
+  Promise<void> addJSModuleResolver(/*FIXME*/, boolean onlyForFrameScript);
```

Comment:
It's for this: https://tc39.github.io/ecma262/#sec-hostresolveimportedmodule

When, within the page, the user does something like; `import x from 'abc://foo'`, the embedder can resolve `abc://foo`, making it possible to expose features to the page.

This will how we will give access to special APIs.